olp/utils.py

```diff
@@ -83,6 +83,29 @@ def remove_perm(user, permission, obj=None):
     return True
 
 
+def remove_permission_for_object(model_instance):
+    """
+    Removes a permission for a given object.
+    """
+
+    from django.contrib.contenttypes.models import ContentType
+    from olp.models import ObjectPermission
+
+    # Get the content type for the given instance
+    object_content_type = ContentType.objects.get_for_model(
+        model_instance.__class__
+    )
+
+    # Unlink any permissions associated with this model
+    permissions = ObjectPermission.objects.filter(
+        base_object_id=model_instance.id,
+        base_object_ct=object_content_type
+    )
+
+    if permissions.count():
```

Comment:
If you call `QuerySet.delete()`, no objects will be removed if the queryset doesn't match any. No error will be raised either, which makes it easier (and less prone to race conditions) to just call `.delete()` instead of checking ahead of time for objects.

Also, you should use `QuerySet.exists()` to check if a queryset has any objects (and you won't be using them) or `len(QuerySet)` if you will be using the objects.